package/pyproject.toml

```diff
@@ -12,3 +12,64 @@ requires = [
     "setuptools",
     "wheel",
 ]
+
+[project]
+name = "MDAnalysis"
+dynamic = ['version',]
+license = {file = "LICENSE"}
+authors = [
+    {name = 'MDAnalysis Development Team', email = 'mdanalysis@numfocus.org'}
+]
+maintainers = [
+    {name = 'MDAnalysis Core Developers', email = 'mdanalysis@numfocus.org'}
+]
+requires-python = ">=3.7"
+dependencies = [
+    'numpy>=1.18.0',
+    'biopython>=1.71',
+    'networkx>=1.0',
```

Comment:
NetworkX 2 is from 2017. Is there a good reason to keep NetworkX 1 supported? Does this prevent the usage of new features in NetworkX 2?

Reply:
Yeah we probably should up all our dependencies, especially as we move closer to NEP29. I'm not against upping the minimum networkx version, I doubt this prevents networkx 2 though - in fact our CI runners use it: https://github.com/MDAnalysis/mdanalysis/runs/5754334356?check_suite_focus=true#step:6:94

Reply:
Yes, I've only ever used NetworkX 2 with MDA I think, that why I'm wondering why NetworkX 1 is still around. My comment was about adding new features requiring NetworkX; if version 1 is kept, one would have to make sure the the change is backward-compatible (and maybe can't use a fancy new function or a function with different API).

Reply:
I see what you mean, tbh I'd be up for just getting rid of networkx as a core dependency overall (I think we barely use it for some leaflet stuff). But yeah, we should bump it up to 2.0+

Reply:
If we ever get a "lipids MDAKit" then we can remove the leafletfinder together with the networkx dependency. However, networkx might be useful for the GSOC project on "ring groups". 

In anhy case, bumping the minimum version really should be fine.